lib/wallet/txdb.js

```diff
@@ -2258,7 +2258,6 @@ class TXDB {
       ns.applyState(delta);
 
       if (ns.isNull()) {
-        await this.removeNameMap(b, nameHash);
```

Comment:
Can you explain this line?

Reply:
NameMap holds a Set of wallet IDs (uint32 values). This Map tells the wallet if the **name** (using names hash as key) is being watched by a particular wallet.

This line if present, removes the wid from nameMap of related name and prevents the wallet to watch the name.

Below is the dump of a signaling scenario to further eloborate,  the wallet **backup** (wid -> 1) sends and Open to name '**ckarlg**' and the namemap of name '**ckarlg**' contains the wid -> 1

Open Request:

```
{
    "hash": "013ce40fb4be66cf5313d614deab8b47de6673ccbfcde9a516e9ef440354727a",
    "witnessHash": "ed12ad83dd8eddf95a6bf8d50898cefd82c25117c29bea58de7ccc7fcfb26e52",
    "mtime": 1621271561,
    "version": 0,
    "inputs": [
        {
            "prevout": {
                "hash": "4bb22f9a400ecc7a98a1c7979b19d25dec04384d6c5a826d9e0d657e95ca81bc",
                "index": 0
            },
            "witness": [
                "1dfa0a52906389524a39d24fe7485d97633525bbbae10f19658463c8bbae82ca1b80adb2c7ca4f208bec3ecd73b2854975ba8a906e23245256b24542da32f04901",
                "03635d59d3ce9bb0c65556696d193de4c644378938ee7fcbd5216f035470451592"
            ],
            "sequence": 4294967295,
            "address": "rs1qgncfk8pnr98zr5ecdxplv33vqv8udtnduwq7ry"
        }
    ],
    "outputs": [
        {
            "value": 0,
            "address": "rs1lqqqqke46ek",
            "covenant": {
                "type": 2,
                "action": "OPEN",
                "items": [
                    "d3ef9e3055f27937b6c2dc35e0537463de8e371bcdb99b76bc27674d2904217c",
                    "00000000",
                    "636b61726c67"
                ]
            }
        },
        {
            "value": 1999996660,
            "address": "rs1q29msm4xxqphuw5cfk28evuua9aypus8k7ty0jj",
            "covenant": {
                "type": 0,
                "action": "NONE",
                "items": []
            }
        }
    ],
    "locktime": 0,
    "hex": "00000000014bb22f9a400ecc7a98a1c7979b19d25dec04384d6c5a826d9e0d657e95ca81bc00000000ffffffff0200000000000000001f020000020320d3ef9e3055f27937b6c2dc35e0537463de8e371bcdb99b76bc27674d2904217c040000000006636b61726c67f486357700000000001451770dd4c6006fc75309b28f96739d2f481e40f600000000000002411dfa0a52906389524a39d24fe7485d97633525bbbae10f19658463c8bbae82ca1b80adb2c7ca4f208bec3ecd73b2854975ba8a906e23245256b24542da32f049012103635d59d3ce9bb0c65556696d193de4c644378938ee7fcbd5216f035470451592"
}
```

leveldb dump (wallets and NameMap for name 'ckarlg') only.


```
wallet: backup, Id: 1
wallet: primary, Id: 0
NameHash: d3ef9e3055f27937b6c2dc35e0537463de8e371bcdb99b76bc27674d2904217c, WIDs: [1] - namemap
```

Querying domainnamestatus, as expected returns the proper state of the auction

```
{
    "name": "ckarlg",
    "nameHash": "d3ef9e3055f27937b6c2dc35e0537463de8e371bcdb99b76bc27674d2904217c",
    "state": "OPENING",
    "height": 5,
    "renewal": 5,
    "owner": {
        "hash": "0000000000000000000000000000000000000000000000000000000000000000",
        "index": 4294967295
    },
    "value": 0,
    "highest": 0,
    "data": "",
    "transfer": 0,
    "revoked": 0,
    "claimed": 0,
    "renewals": 0,
    "registered": false,
    "expired": false,
    "weak": false,
    "stats": {
        "openPeriodStart": 5,
        "openPeriodEnd": 56,
        "blocksUntilBidding": 45,
        "hoursUntilBidding": 7.5
    }
}
```

To be exactly sure we would need to ask Zippy but my guess is: 

When a reorg happens we roll back, since the name is bound to a **null address** instead of wallets own addresses, to not to lose the state `that wallet that opened the name is tracking`, he might preferred **not to** remove the nameMap.